src/Illuminate/Database/Query/Builder.php

```diff
@@ -1667,6 +1664,36 @@ public function aggregate($function, $columns = array('*'))
 	}
 
 	/**
+	 * Create a query to compute an aggregate value.
+	 *
+	 * @param  string  $function
+	 * @param  array   $columns
+	 * @return mixed
+	 */
+	protected function aggregateQuery($function, $columns = array('*'))
+	{
+		$previousColumns = $this->columns;
+
+		// If the developer specified a column to calculate the aggregate for, we will
+		// make sure it is also selected from within the subquery otherwise the data
+		// would not be available when the aggregate is actually being calculated.
+		if ($columns != array('*'))
+		{
+			$this->addSelect($columns);
+
+			$this->columns = array_unique($this->columns);
```

Comment:
If the column was already selected, we need to filter it out here. This is tested in `testAggregateResetFollowedBySelectGet`